### ISSUE TITLE
Online Co-op: Null ref: T2.QuasiModalCanvas.SetCreatureBody_Part2

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#722]
Auto reports from 0.2.0. Online Co-op.

Magic Mirror issues with room transitions...?

Probably was a weird scenario of room transitions occurring during 0.2.0... but online room transitions have been fixed in 0.2.1.

Will keep an eye out to see if this reappears in 0.2.1 or higher.

```
9:45:58, Frame 1659, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.QuasiModalCanvas.SetCreatureBody_Part2 (T2.CreatureBodyID bodyID, T2.CreatureStruct creatureStruct)
T2.NetworkMessage_SetCreatureBody2.ProcessReceivedPacket (T2.NetworkPeer recipient, LiteNetLib.NetPeer sender, LiteNetLib.Utils.NetDataReader reader, T2.GameStruct gameStruct)
T2.NetworkPeer.OnNetworkReceive (LiteNetLib.NetPeer peerThatSentMessage, LiteNetLib.Utils.NetDataReader reader)
LiteNetLib.NetManager.ProcessEvent (LiteNetLib.NetManager+NetEvent evt)
LiteNetLib.NetManager.PollEvents ()
T2.NetworkClient.Update ()

GameStates: ShowingTitleScreen, UsingMainMenu, ViewingAlertBox, TransitioningToDungeonRoom
SystemPlayer InputDevice: 键盘
HumanPlayer 1 InputDevice: 键盘
```

**Comment by ChrisNZL** [comment]
>Will keep an eye out to see if this reappears in 0.2.1 or higher.

**Comment by ChrisNZL** [comment]
Re-opening. 0.2.2 auto report. Feedback ID: 20201230-CCHGD

```
10:33:35, Frame 8512, LOG »  NETWORK HANDSHAKE: Connected.
Device is online.
Server: Lightning-Beijing-01
GameID: NRHQ6


10:33:36, Frame 8561, LOG »  NetworkMessage_DungeonRoomTransitionHelper.TellServerPlayerHasFadedToBlack:
Sending message to server...



10:33:36, Frame 8563, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.QuasiModalCanvas.SetCreatureBody_Part2 (T2.CreatureBodyID bodyID, T2.CreatureStruct creatureStruct)
T2.NetworkMessage_SetCreatureBody2.ProcessReceivedPacket (T2.NetworkPeer recipient, LiteNetLib.NetPeer sender, LiteNetLib.Utils.NetDataReader reader, T2.GameStruct gameStruct)
T2.NetworkPeer.OnNetworkReceive (LiteNetLib.NetPeer peerThatSentMessage, LiteNetLib.Utils.NetDataReader reader)
LiteNetLib.NetManager.ProcessEvent (LiteNetLib.NetManager+NetEvent evt)
LiteNetLib.NetManager.PollEvents ()
T2.NetworkClient.Update ()

GameStates: ShowingTitleScreen, UsingMainMenu, TransitioningToDungeonRoom
OnlineGameInfo: Device is online.
Server: Lightning-Beijing-01
GameID: NRHQ6
SystemPlayer InputDevice: 键盘
HumanPlayer 1 InputDevice: 键盘
```

**Comment by ChrisNZL** [comment]
Addressed in 0.2.2a.

I am not sure what parts of this method are causing the null refs.

[Created a gist here.](https://gist.github.com/ChrisNZL/42e6be8a0227dc5a4bdf627a7afc03ea)

As such, I am now wrapping a lot of this into try/catches with warnings.

Client is not finding something that the server thinks the client should know about... but I am unable to tell what else can be null-checked apart from `creatureStruct` and `creature` at the start. Other things in the method should exist, unless Unity is pulling some weird "object exists but doesn't actually exist" stunt.

In any case, this is just regarding the Magic Mirror after joining a game... Suspect client is receiving a message about character bodies but hasn't received latest game info.
